extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessorRowBase.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.processor;
+
+import io.deephaven.chunk.ObjectChunk;
+import io.deephaven.chunk.WritableChunk;
+
+import java.util.List;
+
+/**
+ * A specialization that {@link #processAll(ObjectChunk, List) processes all} one row at a time.
+ *
+ * <p>
+ * In particular, this is a useful construct when {@code T} is a {@link java.nio.ByteBuffer} or {@code byte[]} type and
+ * it makes sense to parse straight from bytes into output chunks (as opposed to parsing into a more intermediate state
+ * and then processing into chunks in a column-oriented fashion).
+ *
+ * @param <T> the object type
+ */
+public abstract class ObjectProcessorRowBase<T> implements ObjectProcessorRowLimited<T> {
```

Comment:
I'm not sure this abstraction will actually be used; thinking more about how the raw avro handler was written, it first casts all of the writeable chunks and then proceeds row-by-row. As written, the implementations here will need to cast once per row / cell, and that's probably not desirable.